README.md

```diff
@@ -1283,12 +1283,12 @@ _You can enable the following settings in Xcode by running [this script](resourc
     ```swift
     // WRONG
 
-    // In this method, the `newContext` parameter is unused.
+    // In this method, the `newCondition` parameter is unused.
     // This is actually a logical error, and is easy to miss, but compiles without warning.
-    func withContext(_ newContext: Context) {
-      var updatedValue = self
-      updatedValue.context = context
-      return updatedValue  
+    func updateWeather(_ newCondition: WeatherCondition) -> Weather {
+      var updatedWeather = self
+      updatedWeather.condition = condition // this mistake inadvertently makes this method unable to change the weather condition
+      return updatedWeather
     }
```

Comment:
Note that I added the missing return type here! `-> Weather`

the full code of this Weather example is:

```swift
import SwiftUI

struct WeatherView: View {

  @State private var currentWeather = Weather(condition: .sunny)

  var body: some View {
    VStack {
      weatherIcon(type: currentWeather.condition)
        .font(.largeTitle)

      Button("change weather") {
        currentWeather = currentWeather.updateWeather(.rainy)
      }
    }
  }

  private func weatherIcon(type: Weather.WeatherCondition) -> some View {
    switch type {
    case .sunny:
      return Image(systemName: "sun.max.fill")
    case .rainy:
      return Image(systemName: "cloud.rain.fill")
    }
  }
}

struct Weather {
  var condition: WeatherCondition

  enum WeatherCondition {
    case sunny
    case rainy
  }

  func updateWeather(_: WeatherCondition) -> Weather {
    var updatedWeather = self
    updatedWeather.condition = condition
    return updatedWeather
  }
}

struct WeatherView_Previews: PreviewProvider {
  static var previews: some View {
    WeatherView()
  }
}

```